components/datadog/apps/fakeintake/fargateFakeintakeService.go

```diff
@@ -79,6 +85,89 @@ func NewECSFargateInstance(e aws.Environment, option ...fakeintakeparams.Option)
 			return nil, err
 		}
 
+		ca, err := acmpca.NewCertificateAuthority(e.Ctx, namer.ResourceName("ca"), &acmpca.CertificateAuthorityArgs{
+			Type: pulumi.StringPtr("ROOT"),
+			CertificateAuthorityConfiguration: acmpca.CertificateAuthorityCertificateAuthorityConfigurationArgs{
+				KeyAlgorithm:     pulumi.String("RSA_4096"),
+				SigningAlgorithm: pulumi.String("SHA512WITHRSA"),
+				Subject: acmpca.CertificateAuthorityCertificateAuthorityConfigurationSubjectArgs{
+					CommonName: e.CommonNamer.DisplayName(64, pulumi.String("fakeintake CA")),
+				},
+			},
+			RevocationConfiguration: acmpca.CertificateAuthorityRevocationConfigurationArgs{
+				CrlConfiguration: acmpca.CertificateAuthorityRevocationConfigurationCrlConfigurationArgs{
+					Enabled: pulumi.BoolPtr(false),
+				},
+				OcspConfiguration: acmpca.CertificateAuthorityRevocationConfigurationOcspConfigurationArgs{
+					Enabled: pulumi.Bool(false),
+				},
+			},
+			PermanentDeletionTimeInDays: pulumi.IntPtr(7),
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		current, err := paws.GetPartition(e.Ctx, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+		caCert, err := acmpca.NewCertificate(e.Ctx, namer.ResourceName("ca-cert"), &acmpca.CertificateArgs{
+			CertificateAuthorityArn:   ca.Arn,
+			CertificateSigningRequest: ca.CertificateSigningRequest,
+			SigningAlgorithm:          pulumi.String("SHA512WITHRSA"),
+			TemplateArn:               pulumi.String(fmt.Sprintf("arn:%v:acm-pca:::template/RootCACertificate/V1", current.Partition)),
+			Validity: acmpca.CertificateValidityArgs{
+				Value: pulumi.String("10"),
+				Type:  pulumi.String("YEARS"),
+			},
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, err = acmpca.NewCertificateAuthorityCertificate(e.Ctx, namer.ResourceName("ca-cert-cert"), &acmpca.CertificateAuthorityCertificateArgs{
+			CertificateAuthorityArn: ca.Arn,
+			Certificate:             caCert.Certificate,
+			CertificateChain:        caCert.CertificateChain,
+		}, opts...); err != nil {
+			return nil, err
+		}
+
+		domainName := alb.LoadBalancer.DnsName().ApplyT(func(dnsName string) *string {
+			if len(dnsName) <= 64 {
+				return &dnsName
+			}
+			dummyDNSName := "fakeintake.datadoghq.com"
+			return &dummyDNSName
+		}).(pulumi.StringPtrOutput)
+		cert, err := acm.NewCertificate(e.Ctx, namer.ResourceName("cert"), &acm.CertificateArgs{
+			CertificateAuthorityArn: ca.Arn,
+			KeyAlgorithm:            pulumi.String("RSA_2048"),
+			DomainName:              domainName,
+			SubjectAlternativeNames: pulumi.StringArray{
+				alb.LoadBalancer.DnsName(),
+			},
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, err = clb.NewListener(e.Ctx, namer.ResourceName("lb-https"), &clb.ListenerArgs{
+			LoadBalancerArn: alb.LoadBalancer.Arn(),
+			Port:            pulumi.IntPtr(sslPort),
+			Protocol:        pulumi.StringPtr("HTTPS"),
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Fascinating, can we have a listener without a load balancer ?

Reply:
Looks like we cannot https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html

Reply:
A standalone listener makes no sense if not attached to a load-balancer.
But a load-balancer can have several listeners.
That’s why, on AWS side, they are distinct objects.

The `awsx` pulumi module provides some syntax sugar to create in one call, the loadbalancer, and its attached target group and listeners. But with the lower level `aws` module, that would have been three calls to create three AWS objects.
This is even visible in the output of pulumi:
```
   │  ├─ awsx:lb:ApplicationLoadBalancer                                aws-fakeintake-lb                                 created (7s)        
   │  │  ├─ aws:lb:TargetGroup                                          aws-fakeintake-lb                                 created (4s)        
   │  │  ├─ aws:lb:LoadBalancer                                         aws-fakeintake-lb                                 created (17s)      
   │  │  └─ aws:lb:Listener                                             aws-fakeintake-lb-0                               created (5s)        
```

To create an HTTPS listener we have no choice but splitting things because:

* In order to create the HTTPS LB listener, we need an SSL certificate.
* In order to create the SSL certificate, we need to know the DNS name to put in it.
* And we need to create the load balancer to gets its DNS name.